### DISCUSSION TITLE
Changing snippet trigger length to 1 instead of 3

### DISCUSSION THREAD

**Discussion by mikeynic** [VonHeikemen/lsp-zero.nvim#87] [Q&A] [answered]
When using the LSP, I have to type 3 letters in order to trigger the autocompletion. I was wondering if there is a way to change it to trigger after just typing 1 letter?

I think the code responsible is this part of the cmp_config:

`  
  sources = {
    {name = 'path'},
    {name = 'nvim_lsp', keyword_length = 3},
    {name = 'buffer', keyword_length = 3},
    {name = 'luasnip', keyword_length = 2},
  }
`

Was wondering where this file is located / what I should add to my config to change it?

Edit: Sorry if this is a really basic question, I'm really new to all of this. And thanks so much for this plug-in! It's made setting up all of this so much easier.

**Comment by VonHeikemen** [comment, marked as answer]
Right now I don't have a good way to override the configuration for sources. What I tell people is to use `.setup_nvim_cmp`, copy the defaults in the `sources` options, and modify them to fit their needs.

```lua
local lsp = require('lsp-zero')
lsp.preset('recommended')

lsp.setup_nvim_cmp({
  sources = {
    {name = 'path'},
    {name = 'nvim_lsp', keyword_length = 3},
    {name = 'buffer', keyword_length = 3},
    {name = 'luasnip', keyword_length = 2},
  }
})

lsp.setup()
```

**Reply by mikeynic** [reply]
Thanks very much for the response. That helps a lot I'll try that out now!
